src/app-service.ts

```diff
@@ -228,10 +228,13 @@ export class AppService extends EventEmitter {
         try {
             await possiblePromise;
             res.send({});
-        } catch (e) {
+        } catch (e: any) {
+            const parsed = parseInt(e.status);
+            const status = isNaN(parsed) ? 400 : parsed;
```

Comment:
We should probably default to a 500 error, which is the general purpose "something went wrong on the backend" which is all we can assume if the error isn't structured. A 400 would give the impression that it was a client-side issue, which we don't know for sure.

Reply:
Agree. Will change to 500.